src/generic_core/remote-instance.ts

```diff
@@ -195,19 +189,20 @@ export var remoteProxyInstance :RemoteInstance = null;
           // TODO: Move the logic for resetting the onceSharerCreated promise inside
           // remote-connection.ts.
           this.connection_.resetSharerCreated();
-          this.onceSharerReadyForOffer_ = this.connection_.onceSharerCreated;
+          this.hasSeenOffer = true;
```

Comment:
The logic relies on the client peer not being able to try to start getting if they are already getting (and therefore the sharer is already sharing). This means the sharer can only get an offer if their SharingState is NONE or TRYING_TO_SHARE_ACCESS; so it should never happen that hasSeenOffer is set to true in handleSignal and then set back to false in stopShare. The resetting in stopShare should only happen if the peer clicked "Stop Getting Access"